responses/__init__.py

```diff
@@ -323,13 +327,21 @@ def _url_matches(self, url, other, match_querystring=False):
                 url = _clean_unicode(url)
                 if not isinstance(other, six.text_type):
                     other = other.encode("ascii").decode("utf8")
+
             if match_querystring:
-                return self._url_matches_strict(url, other)
+                rfc_compliant_url = parse_url(url).url
+                rfc_result = self._url_matches_strict(rfc_compliant_url, other)
+                non_rfc_result = self._url_matches_strict(url, other)
```

Comment:
Are there scenarios where the RFC compliant matching fails but the existing behavior does match?

Reply:
Not that I can think of however, it should not make a difference. The final returned result is the `or` between the RFC match and the existing behavior. If say the RFC match failed but the existing behavior did succeed, then it would return `True` regardless.

Reply:
Do we need the non-rfc compliant host comparisons then? Seems like the RFC comparisons are more relaxed than what we had before.

Reply:
If this behavior is acceptable than no, I don't think the non-RFC comparison is needed.

Reply:
I think it is fine. The case insensitive comparisons are how HTTP requests would work in the wild.